evi-react-native-example/modules/audio/ios/SoundPlayer.swift

```diff
@@ -0,0 +1,102 @@
+import ExpoModulesCore
+import AVFoundation
+import Foundation
+
+// This is a simple sound player that uses the AVAudioPlayer API to supports playing a single audio
+// clip at a time. Maintaining a queue of pending audio clips is left to the user (to be done
+// in Javascript).
+public class SoundPlayer {
+    private var lastAudioPlayer: AVAudioPlayer?
+    public func stopPlayback() {
+        self.lastAudioPlayer?.stop()
+        self.lastAudioPlayer = nil
+    }
+
+    // This method takes a chunk of base64-encoded audio and plays it via the AVAudioPlayer API.
+    // It uses `withUnsafeThrowingContinuation` so that on the Javascript side it will return a promise
+    // that resolves when the audio is finished playing.
+    public func playBase64Audio(_ base64String: String) async throws {
+        logAudioSessionInfo()
+        // It's important to define `delegate` in this scope, as `audioPlayer.delegate` is a weak reference and this will be deallocated if it is only
+        // defined within the inner scope of `withUnsafeThrowingContinuation`.
+        var delegate: ContinuationDelegate!
+        var audioPlayer: AVAudioPlayer!
+        return try await withUnsafeThrowingContinuation { continuation in
+            do {
+                guard let data = Data(base64Encoded: base64String) else {
+                    continuation.resume(throwing: NSError(domain: "AudioModule", code: 1, userInfo: [NSLocalizedDescriptionKey: "Invalid base64 string"]))
+                    return
+                }
+                audioPlayer = try AVAudioPlayer(data: data, fileTypeHint: AVFileType.wav.rawValue)
+                self.lastAudioPlayer = audioPlayer
+                delegate = ContinuationDelegate(continuation: continuation)
+                audioPlayer.delegate = delegate
```

Comment:
`AVAudioPlayer.delegate` is [weakly held](https://developer.apple.com/documentation/avfaudio/avaudioplayer/1387134-delegate), so once we leave execution of `playBase64Audio(_:)`, it should be nil if i'm following right, even though you persist the audioPlayer. Is that expected? if not, i'd move the delegate to be owned by the SoundPlayer instance rather than the func